src/test/java/com/puppycrawl/tools/checkstyle/api/XpathMapperTest.java

```diff
@@ -0,0 +1,151 @@
+package com.puppycrawl.tools.checkstyle.api;
+
+import antlr.RecognitionException;
+import antlr.TokenStreamException;
+import com.puppycrawl.tools.checkstyle.BaseCheckTestSupport;
+import com.puppycrawl.tools.checkstyle.TreeWalker;
+import com.puppycrawl.tools.checkstyle.xpath.ElementNode;
+import com.puppycrawl.tools.checkstyle.xpath.RootNode;
+import net.sf.saxon.om.GenericTreeInfo;
+import net.sf.saxon.om.Item;
+import net.sf.saxon.sxpath.XPathDynamicContext;
+import net.sf.saxon.sxpath.XPathEvaluator;
+import net.sf.saxon.sxpath.XPathExpression;
+import net.sf.saxon.trans.XPathException;
+import org.junit.Test;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.List;
+
+import static org.junit.Assert.assertArrayEquals;
+
+/**
+ * Created by timur on 6/13/17.
+ */
+public class XpathMapperTest extends BaseCheckTestSupport {
+
+    @Test
+    public void testOne() throws XPathException, IOException, TokenStreamException, RecognitionException {
+        final FileText text = new FileText(new File("src/test/resources/com/puppycrawl/tools/checkstyle/api/TestAst.java"),
+                System.getProperty("file.encoding", "UTF-8"));
+        final FileContents contents = new FileContents(text);
+
+        DetailAST rootAST = TreeWalker.parse(contents);
+        DetailNode detailNode = null;
+
+        RootNode rootNode = new RootNode();
+        ElementNode classDefinition = new ElementNode(TokenTypes.CLASS_DEF, rootNode, rootNode, detailNode, rootAST.getNextSibling());
+
+        XPathEvaluator xpathEvaluator = new XPathEvaluator();
+        String xpath = "/CLASS_DEF[@Image='TestAst']";
```

Comment:
@romani @rnveach @timurt 
I'm not OK with the name of the attribute. What does it mean? DetailAST has nothing in common with Image. It is not the class field or getter/setter. If we want to refer to a class with the name `TestAst` the xpath-query should be:
```
//CLASS_DEF[@Name='TestAst']
```

or

```
//CLASS_DEF[@Text='TestAst']
```

as actually we refer to the field with the name text which is a part of CommonAST class.